### DISCUSSION TITLE
Support for Pest tests

### DISCUSSION THREAD

**Discussion by faustbrian** [stefanzweifel/laravel-stats#194] [Ideas]
Currently the whole evaluation is designed around classes which are loaded by https://github.com/stefanzweifel/laravel-stats/blob/13474eacf12a2ae06cf1be3652bd8fdde96df9dd/src/ClassesFinder.php#L18-L33 which makes it difficult or maybe even impossible to currently add support for Pest tests which use functions like `it` and `test` instead of using classes like PHPUnit which means the code to test ratio will be wrong because it doesn't detect the Pest tests.

Supporting this would probably require quite a bit of change because we would need to handle classes and some kind of pattern matching inside of files to see if we are dealing with a pest file if we are encountering a file that ends with `Test.php`.

Would this be a change for which a PR would be accepted or would this be too big a of change and not worth the effort just to support Pest? I've already tried to detect pest tests with the current system but couldn't make it work because everything is designed around classes.

**Comment by stefanzweifel** [comment]
As you mentioned, everything is designed around classes as I never thought something like Pest would exist 3-4 years ago. 😅 

I think adding support for Pest would definitely be great. It's getting more and more traction and I think it's a great project. (I haven't used it myself in a project, but plan to do so very soon)

I would glady accept a well thought out PR. But as you mentioned, you would have to change quite a lot here. (I would even say that it's basically a rewrite of the core of the package.)

In #153 I hinted, that I think it would be great if we could expand `laravel-stats` beyond the Laravel ecosystem and open it open to all PHP projects. If we do that, we definitely need support for PHP files which just contains functions or are not classes.


If you have any ideas on how to tackle these problems, feel free to submit a PR or let's discuss it here.

**Reply by faustbrian** [reply]
Don't have any concrete ideas yet but will try to play with some over the next days and see what works best without adding too much overhead.
